### PR TITLE
Indicate publiclink access

### PR DESCRIPTION
**Pull request by kobergj**
Indicates if a user in the context is coming through a public link

### CODE REVIEW

File: pkg/auth/manager/ocmshares/ocmshares.go
Comment (2403905) [inline pkg/auth/manager/ocmshares/ocmshares.go:162]: is the "impersonating-user" suitable for an anonymous user?

Comment (kobergj) [reply pkg/auth/manager/ocmshares/ocmshares.go:162]: Not sure what you mean.

"impersonating-user" can be 
* a hardcoded struct when this is a public link
* a remote user when this is a federate share

Comment (2403905) [reply pkg/auth/manager/ocmshares/ocmshares.go:162]: Nevermind. This is just a naming nitpick.

**Review by 2403905 — APPROVED** [review]

**Comment by kobergj** [comment]
I'll wait for green ocis pipeline before merge https://github.com/owncloud/ocis/pull/10116  But no more changes expected

**Pull request merged**
